tests/src/Kernel/ConfigEntityTest.php

```diff
@@ -66,6 +76,46 @@ public function testConfigAction() {
   }
 
   /**
+   * Test saving a deeply nested rule.
+   *
+   * @todo This test currently fails if $strictConfigSchema = TRUE.
+   */
+  public function testNestedRuleSave() {
+    // This test is specific to reaction rules, so get the appropriate
+    // entity storage:
+    $storage = $this->container->get('entity.manager')->getStorage('rules_reaction_rule');
+    // Create a nested expression.
+    $and = $this->expressionManager->createAnd()
+      ->addCondition('rules_test_false')
+      ->addCondition('rules_test_true', ContextConfig::create()->negateResult())
+      ->negate();
+
+    $or = $this->expressionManager->createOr()
+      ->addCondition('rules_test_true', ContextConfig::create()->negateResult())
+      ->addCondition('rules_test_false')
+      ->addCondition($and);
+
+    $rule = $this->expressionManager->createRule();
```

Comment:
createReactionRule()  - the plugin does not fit the config entity else ;) I guess we should check the class somewhere in the config enttiy impl?